logback/src/main/java/com/linecorp/armeria/common/logback/RequestContextExportingAppender.java

```diff
@@ -181,8 +182,22 @@ private void ensureNotStarted() {
         }
     }
 
+    /**
+     * It is possible that requestContextStorageProvider hasn't been initialized yet
+     * due to static variable circular dependency.
+     * Most notably, this can happen when logs are appended while trying to initialize
+     * Flags#requestContextStorageProvider.
+     */
+    private static boolean isInitialized() {
+        return Flags.initialized();
+    }
+
     @Override
     protected void append(ILoggingEvent eventObject) {
+        if (!isInitialized()) {
+            return;
```

Comment:
How about delegating the event as-is so it's not lost during initialization?
```suggestion
            aai.appendLoopOnAppenders(eventObject);
            return;
```

Reply:
Thanks for catching this 👍  Done